From 70a61653d2bbab8a8e4cadac73bfb72eeee91b24 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Mon, 24 May 2021 17:15:39 +0300
Subject: [PATCH] [SW-43147] gaudi: ras: set btl index in dev sts

Update BTL index in RAS device status opcode, as
it was previously missing.

Change-Id: I69b361dce26bd589e16efc6fd8c9755975c7caba
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 84b94d4..340aad1 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -423,3 +423,10 @@
 	regs->gic_nic_qm_irq_ctrl = mmGIC_NIC_QM_IRQ_CTRL_POLL_REG;
 	regs->gic_dma_core_irq_ctrl = mmGIC_DMA_CR_IRQ_CTRL_POLL_REG;
 }
+
+u32 hl_get_btl_index(void)
+{
+	u32 btl_id = (regmap_readl(mmBTL_ID) >> 8) & 0xff;
+
+	return (btl_id & 0x3) - 1;
+}
diff --git a/preboot/i2cslave_indirect.c b/preboot/i2cslave_indirect.c
index aa37f48..3b9da37 100644
--- a/preboot/i2cslave_indirect.c
+++ b/preboot/i2cslave_indirect.c
@@ -493,6 +493,11 @@
 	err_chksum_avail = true;
 }
 
+__weak u32 hl_get_btl_index(void)
+{
+	return 0;
+}
+
 static void i2csi_alert_set_pcie(void)
 {
 	if (!hl_pci_link_is_up())
@@ -512,6 +517,7 @@
 		req_rd.length = sizeof(op_info);
 		break;
 	case HL_I2CI_REQ_OP_DEVICE_STATUS:
+		op_status.used_btl_image_index = hl_get_btl_index();
 		req_rd.data = (u8 *)&op_status;
 		req_rd.length = sizeof(op_status);
 		break;
